### PR TITLE
fix(flags): do not attempt to load usage tab for deleted flags

### LINKED ISSUE #24454: Better handling of deleted flags
## Is your feature request related to a problem?
We currently soft-delete feature flags by setting the `deleted` field to `true`. Such flags:
- can be looked up with an URL
- confusingly may show "Enabled" in the UI
- when requested, still show `Feature flag events` in the usage tab (even though they're not billed)

## Describe the solution you'd like
- some banners/tooltips would solve this easily

### Debug info

```shell
- [ ] PostHog Cloud, Debug information: [please copy/paste from https://us.posthog.com/settings/project-details#variables]
- [ ] PostHog Hobby self-hosted with `docker compose`, version/commit: [please provide]
- [ ] PostHog self-hosted with Kubernetes (deprecated, see [`Sunsetting Kubernetes support`](https://posthog.com/blog/sunsetting-helm-support-posthog)), version/commit: [please provide]
```

### PR DESCRIPTION
**Pull request by havenbarnes**
## Problem

Closes https://github.com/PostHog/posthog/issues/24454

## Changes

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

Screenshot 2024-11-11 at 11 34 41 AM
Screenshot 2024-11-11 at 11 24 29 AM

## Does this work well for both Cloud and self-hosted?

Yes

## How did you test this code?

- Added a cypress assertion validating the functionality
- Tested functionality locally

### CODE REVIEW

**Review by dmarticus — DISMISSED** [review]

**Review by dmarticus — COMMENTED** [review]
Actually, I take back my initial stamp, because while this solves the UI issue for the feature flags page, I think we still need to do somewhere with the insights that were created for this feature flag.  What happens when we try and load an auto-generated insight for a flag that's been deleted?

**Review by dmarticus — COMMENTED** [review]
Haven and I chatted about the autogenerated insights stuff offline; here's our convo:

image

TL;DR :shipit:

**Review by dmarticus — APPROVED** [review]
